zuul-core/src/main/java/com/netflix/zuul/origins/OriginName.java

```diff
@@ -44,30 +42,36 @@
      * used for establishing a secure connection, as well as logging.
      */
     private final String authority;
+
     /**
-     * Indicates if the authority comes from a trusted source.
+     * @deprecated use {@link #fromVipAndApp(String, String)}
      */
-    private final boolean authorityTrusted;
-
+    @Deprecated
     public static OriginName fromVip(String vip) {
-        return fromVip(vip, vip);
+        return fromVipAndApp(vip, VipUtils.extractUntrustedAppNameFromVIP(vip));
     }
 
+    /**
+     * @deprecated use {@link #fromVipAndApp(String, String, String)}
+     */
+    @Deprecated
     public static OriginName fromVip(String vip, String niwsClientName) {
-        return new OriginName(niwsClientName, vip, VipUtils.extractUntrustedAppNameFromVIP(vip), false);
+        return fromVipAndApp(vip, VipUtils.extractUntrustedAppNameFromVIP(vip), niwsClientName);
     }
 
-    @CheckReturnValue
-    public OriginName withTrustedAuthority(String authority) {
-        return new OriginName(niwsClientName, target, authority, true);
+    public static OriginName fromVipAndApp(String vip, String appName) {
+        return fromVipAndApp(vip, appName, vip);
```

Comment:
This isn't obvious without a comment explaining why the vip and niws name may or may not be the same.

Reply:
Followed up here: https://github.com/Netflix/zuul/pull/1067